File: Desktop/epicoders FX NETBEANS/HappJAvaFX/src/tn/happ/Model/Platter.java
package tn.happ.Model;


import java.util.Objects;

public class Platter {
    private int IdPlatter;
    private String NamePlatter;
    private String ImagePlatter;
    private String ingredient;
    private float PricePlatter;
    private int NbPlatter;
    private String DescriptionPlatter;
    private String TypePlatter;
  //  private int ID_USER;
    private User user;

    public Platter() {
    }


    public Platter(int idPlatter, String namePlatter, String imagePlatter, String ingredient, float pricePlatter, int nbPlatter, String descriptionPlatter, String typePlatter, User user) {
        IdPlatter = idPlatter;
        NamePlatter = namePlatter;
        ImagePlatter = imagePlatter;
        this.ingredient = ingredient;
        PricePlatter = pricePlatter;
        NbPlatter = nbPlatter;
        DescriptionPlatter = descriptionPlatter;
        TypePlatter = typePlatter;
        this.user = user;
    }

    public Platter(String namePlatter, String imagePlatter, String ingredient, float pricePlatter, int nbPlatter, String descriptionPlatter, String typePlatter, User user) {
        NamePlatter = namePlatter;
        ImagePlatter = imagePlatter;
        this.ingredient = ingredient;
        PricePlatter = pricePlatter;
        NbPlatter = nbPlatter;
        DescriptionPlatter = descriptionPlatter;
        TypePlatter = typePlatter;
        this.user = user;
    }

    public int getIdPlatter() {
        return IdPlatter;
    }

    public void setIdPlatter(int idPlatter) {
        IdPlatter = idPlatter;
    }

    public String getNamePlatter() {
        return NamePlatter;
    }

    public void setNamePlatter(String namePlatter) {
        NamePlatter = namePlatter;
    }

    public String getImagePlatter() {
        return ImagePlatter;
    }

    public void setImagePlatter(String imagePlatter) {
        ImagePlatter = imagePlatter;
    }

    public String getIngredient() {
        return ingredient;
    }

    public void setIngredient(String ingredient) {
        this.ingredient = ingredient;
    }

    public float getPricePlatter() {
        return PricePlatter;
    }

    public void setPricePlatter(float pricePlatter) {
        PricePlatter = pricePlatter;
    }

    public int getNbPlatter() {
        return NbPlatter;
    }

    public void setNbPlatter(int nbPlatter) {
        NbPlatter = nbPlatter;
    }

    public String getDescriptionPlatter() {
        return DescriptionPlatter;
    }

    public void setDescriptionPlatter(String descriptionPlatter) {
        DescriptionPlatter = descriptionPlatter;
    }

    public String getTypePlatter() {
        return TypePlatter;
    }

    public void setTypePlatter(String typePlatter) {
        TypePlatter = typePlatter;
    }

    public User getUser() {
        return user;
    }


    public void setUser(User user) {
        this.user = user;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Platter platter = (Platter) o;
        return IdPlatter == platter.IdPlatter;
    }

    @Override
    public int hashCode() {
        return Objects.hash(IdPlatter);
    }

    @Override
    public String toString() {
        return "Platter{" +
                "IdPlatter=" + IdPlatter +
                ", NamePlatter='" + NamePlatter + '\'' +
                ", ImagePlatter='" + ImagePlatter + '\'' +
                ", ingredient='" + ingredient + '\'' +
                ", PricePlatter=" + PricePlatter +
                ", NbPlatter=" + NbPlatter +
                ", DescriptionPlatter='" + DescriptionPlatter + '\'' +
                ", TypePlatter='" + TypePlatter + '\'' +
                '}';
    }



}

File: src/Entity/Platter.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * Platter
 *
 * @ORM\Table(name="platter")
 * @ORM\Entity
 */
class Platter
{
    /**
     * @var int
     *
     * @ORM\Column(name="IdPlatter", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idplatter;

    /**
     * @var string
     *
     * @ORM\Column(name="NAMEPLATTER", type="string", length=50, nullable=false)
     */
    private $nameplatter;

    /**
     * @var string
     *
     * @ORM\Column(name="ingredient", type="string", length=50, nullable=false)
     */
    private $ingredient;

    /**
     * @var int
     *
     * @ORM\Column(name="PRICEPLATTER", type="integer", nullable=false)
     */
    private $priceplatter;

    /**
     * @var int
     *
     * @ORM\Column(name="NBPLATTER", type="integer", nullable=false)
     */
    private $nbplatter;

    /**
     * @var string
     *
     * @ORM\Column(name="DESCRIPTIONPLATTER", type="string", length=50, nullable=false)
     */
    private $descriptionplatter;

    /**
     * @var string
     *
     * @ORM\Column(name="TypePlatter", type="string", length=255, nullable=false)
     */
    private $typeplatter;

    /**
     * @var \Doctrine\Common\Collections\Collection
     *
     * @ORM\ManyToMany(targetEntity="Product", mappedBy="fkplatter")
     */
    private $fkproduct;

    /**
     * Constructor
     */
    public function __construct()
    {
        $this->fkproduct = new \Doctrine\Common\Collections\ArrayCollection();
    }

    public function getIdplatter(): ?int
    {
        return $this->idplatter;
    }

    public function getNameplatter(): ?string
    {
        return $this->nameplatter;
    }

    public function setNameplatter(string $nameplatter): self
    {
        $this->nameplatter = $nameplatter;

        return $this;
    }

    public function getIngredient(): ?string
    {
        return $this->ingredient;
    }

    public function setIngredient(string $ingredient): self
    {
        $this->ingredient = $ingredient;

        return $this;
    }

    public function getPriceplatter(): ?int
    {
        return $this->priceplatter;
    }

    public function setPriceplatter(int $priceplatter): self
    {
        $this->priceplatter = $priceplatter;

        return $this;
    }

    public function getNbplatter(): ?int
    {
        return $this->nbplatter;
    }

    public function setNbplatter(int $nbplatter): self
    {
        $this->nbplatter = $nbplatter;

        return $this;
    }

    public function getDescriptionplatter(): ?string
    {
        return $this->descriptionplatter;
    }

    public function setDescriptionplatter(string $descriptionplatter): self
    {
        $this->descriptionplatter = $descriptionplatter;

        return $this;
    }

    public function getTypeplatter(): ?string
    {
        return $this->typeplatter;
    }

    public function setTypeplatter(string $typeplatter): self
    {
        $this->typeplatter = $typeplatter;

        return $this;
    }

    /**
     * @return Collection<int, Product>
     */
    public function getFkproduct(): Collection
    {
        return $this->fkproduct;
    }

    public function addFkproduct(Product $fkproduct): self
    {
        if (!$this->fkproduct->contains($fkproduct)) {
            $this->fkproduct[] = $fkproduct;
            $fkproduct->addFkplatter($this);
        }

        return $this;
    }

    public function removeFkproduct(Product $fkproduct): self
    {
        if ($this->fkproduct->removeElement($fkproduct)) {
            $fkproduct->removeFkplatter($this);
        }

        return $this;
    }

}
